src/lib/named_hosts_text_ui.ml

```diff
@@ -0,0 +1,275 @@
+(**************************************************************************)
+(*    Copyright 2014, 2015, 2016:                                         *)
+(*          Sebastien Mondet <seb@mondet.org>,                            *)
+(*          Leonid Rozenberg <leonidr@gmail.com>,                         *)
+(*          Arun Ahuja <aahuja11@gmail.com>,                              *)
+(*          Jeff Hammerbacher <jeff.hammerbacher@gmail.com>               *)
+(*                                                                        *)
+(*  Licensed under the Apache License, Version 2.0 (the "License");       *)
+(*  you may not use this file except in compliance with the License.      *)
+(*  You may obtain a copy of the License at                               *)
+(*                                                                        *)
+(*      http://www.apache.org/licenses/LICENSE-2.0                        *)
+(*                                                                        *)
+(*  Unless required by applicable law or agreed to in writing, software   *)
+(*  distributed under the License is distributed on an "AS IS" BASIS,     *)
+(*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or       *)
+(*  implied.  See the License for the specific language governing         *)
+(*  permissions and limitations under the License.                        *)
+(**************************************************************************)
+
+open Ketrew_pure
+open Internal_pervasives
+open Unix_io
+
+let log_status =
+  let open Log in
+  function
+  | `Alive `Idle -> s "Alive"
+  | `Alive (`Askpass_waiting_for_input l) ->
+    s "Asking for passowrd input: " % n %
+    indent (separate n @@ List.map l ~f:(fun (t, question) ->
+        s "* " % Time.log t % s ": " % quote question))
+  | `Dead reason -> s "Dead: " % s reason
+  | `Configured -> s "Configured/ready to start"
+  | `Unknown err -> s "Unknown status: " % quote err
+
+let perform_call ~configuration msg f =
+  Client.as_client ~configuration ~f:begin fun ~client ->
+    Client.call_process_holder client msg
+    >>= fun down ->
+    begin match down with
+    | `Error err ->
+      fail (`Failure (fmt "Server answered with an ERROR: %s" err))
+    | other ->
+      begin match f down with
+      | Some s -> return s
+      | None ->
+      fail (`Failure (fmt "Unexpected process-holder down-message"))
+      end
+    end
+  end
```

Comment:
Now I understand what it is doing, but still a bit awkward without understanding the context of the rest of the methods in this module.